### ISSUE TITLE
WDA Crashes on start 

### ISSUE THREAD

**Issue by levank707** [appium/appium#15244]
## The problem
WDA IS CRASHING ON START  (Appium Desktop)

## Environment

* Appium version (or git revision) that exhibits the issue: Appium v1.20.2
* Last Appium version that did not exhibit the issue (if applicable):Appium v1.20.2
* Desktop OS/version used to run Appium: Mac M1
* Node.js version (unless using Appium.app|exe): v15.11.0
* Npm or Yarn package manager: 7.6.0
* Mobile platform/version under test: iOS
* Real device or emulator/simulator: Real Device
* Appium CLI or Appium.app|exe: Appium Desktop

## Link to Appium logs

https://gist.github.com/levank707/08266a300dba8599bb8a375a55fad788

**Comment by levank707** [comment]
Problem is on real device only, on simulator it works fine

**Comment by KazuCocoa** [comment]
Duplicate of https://github.com/appium/appium/issues/15199
Please try out the latest `appium@beta` (WDA 3.5.0+)

**Comment by levank707** [comment]
> Duplicate of #15199
> Please try out the latest `appium@beta` (WDA 3.5.0+)

Kazucocoa, Where can I find WDA 3.5.0+ Version ?

**Comment by KazuCocoa** [comment]
The latest `appium@beta` has it in the `node_module/appium-webdriveragent`

**Comment by levank707** [comment]
> The latest `appium@beta` has it in the `node_module/appium-webdriveragent`

I have downloaded WDA  from here https://github.com/appium/WebDriverAgent  and  Appium desktop beta (1.20.2-1) from here https://github.com/appium/appium-desktop/releases but  its same :/

**Comment by KazuCocoa** [comment]
How did you start the session?
if you installed WDA on your device directly and launched the WDA, then you also can call `http://<your device IP>:<running port>/screenshot` on your browser to check if the issue has been resolved.

**Comment by levank707** [comment]
> How did you start the session?
> if you installed WDA on your device directly and launched the WDA, then you also can call `http://<your device IP>:<running port>/screenshot` on your browser to check if the issue has been resolved.

I didn't install WDA directly , I just build  it via XCODE and after running Appium desktop WDA was installed on device. Kazu I can't figure how to find what is my   running port  on iPhone, : /

**Comment by levank707** [comment]
> How did you start the session?
> if you installed WDA on your device directly and launched the WDA, then you also can call `http://<your device IP>:<running port>/screenshot` on your browser to check if the issue has been resolved.

I did what you said, I call it   in browser and this is result   https://gist.github.com/levank707/7a9899bc0953a3f6fe855a9bc892f37d

**Comment by levank707** [comment]
I tried same case on my second mac (intel) appium desktop version (1.19.1)  + carthage  and there worked everything fine :/

**Comment by KazuCocoa** [comment]
Could you remove all `/Users/<yourname>/Library/Developer/Xcode/DeriveData` once for https://github.com/appium/appium/issues/15244#issuecomment-812582647 ?
I tried my iPhone SE 2nd gen (iOS 14.4.2) and M1 macOS (Xcode 12.3 and 12.5 beta) with latest WDA https://github.com/appium/WebDriverAgent commit (a69b5419cc73c96706083d59eeb43c41e54c1e90). It worked.

**Comment by levank707** [comment]
I did everything you said , but nothing  it is same [WD Proxy] Got response with status 500: {"value":{"error":"unknown error","message":"-[XCUIScreen screenshotDataForQuality:rect:error:]. Ahhh sh*t here we go again (CJ) :D

**Comment by levank707** [comment]
iPhone XR 14.4.1 , M1 macOS, Xcode 12.5 beta 3, latest WDA from your repository

**Comment by KazuCocoa** [comment]
Could you give us the entire error?

Because `[XCUIDevice(FBHelpers) fb_rawScreenshotWithQuality:error:]` in your log ( https://gist.github.com/levank707/7a9899bc0953a3f6fe855a9bc892f37d ) no longer exists in current https://github.com/appium/WebDriverAgent master commit (a69b5419cc73c96706083d59eeb43c41e54c1e90).
The code was removed in https://github.com/appium/WebDriverAgent/pull/470

**Comment by levank707** [comment]
sorry for late, here is a full log. https://gist.github.com/levank707/b7c62cadfb8df3ef9db30a5e80fd91fd

**Comment by KazuCocoa** [comment]
Thanks.
I see, you tried out 1.20.2. The error message also the old WDA's one.

You should use `appium@beta`. https://github.com/appium/appium-desktop#connect-to-local-appium-server is the way to attach to beta from your Appium Desktop.

**Comment by levank707** [comment]
> Thanks.
> I see, you tried out 1.20.2. The error message also the old WDA's one.
> 
> You should use `appium@beta`. https://github.com/appium/appium-desktop#connect-to-local-appium-server is the way to attach to beta from your Appium Desktop.

Kazu, I did what you said  and there was the same error here is full log 

https://gist.github.com/levank707/c9c135b6066b9b3ce12b02c78c39410c

**Comment by KazuCocoa** [comment]
Thanks.
The below error is newer than https://github.com/appium/appium/issues/15244#issuecomment-812779653 but probably not the latest which includes our screenshot fixes for newer env. A method called in this error, `takeWithScreenID:scale:compressionQuality:rect:uti:error:`, was a bit changed in the latest one.

```
[WD Proxy] Got response with status 500: {"value":{"error":"unknown error","message":"-[XCUILocalDeviceScreenDataSource requestScreenshotOfScreenWithID:withRect:scale:formatUTI:compressionQuality:withReply:]: unrecognized selector sent to instance 0x280cc8bc0","traceback":"(\n\t0   CoreFoundation                      0x00000001904c5880 727F2644-EB4E-3D57-BC2E-E6803BA92366 + 1202304\n\t1   libobjc.A.dylib                     0x00000001a54dec50 objc_exception_throw + 60\n\t2   CoreFoundation                      0x00000001903cc95c 727F2644-EB4E-3D57-BC2E-E6803BA92366 + 182620\n\t3   CoreFoundation                      0x00000001904c8438 727F2644-EB4E-3D57-BC2E-E6803BA92366 + 1213496\n\t4   CoreFoundation                      0x00000001904ca740 _CF_forwarding_prep_0 + 96\n\t5   WebDriverAgentLib                   0x00000001068a5d24 +[FBScreenshot takeWithScreenID:scale:compressionQuality:rect:uti:error:] + 468\n\t6   WebDriverAgentLib                   0x00000001068a5698 +[FBScreenshot takeWithQuality:error:] + 384\n\t7   WebDriverAgentLib
```

What version of wda did you use?
`/opt/homebrew/lib/node_modules/appium/node_modules/appium-webdriveragent/package.json` should have `version` 3.7.1 if you had the latest modules. If no, you could uninstall `appium` once and re-install `appium@beta` to get rest latest once.

**Comment by levank707** [comment]
Kazu thanks not me but you very much  for helping me.  So I checked and   It's 
  "version": "3.4.0". Okay I'll uninstall appium and re-install it

**Comment by levank707** [comment]
So, I did everything, now version is 3.7.1.  but when I am trying to attach appium desktop to appium@beta  WDA is not installing on the device here is error  "An unknown server-side error occurred while processing the command. Original error: Unable to launch WebDriverAgent because of xcodebuild failure: Unable to start WebDriverAgent: Error: spawn xcodebuild ENOENT" 

But I also tried to run my test project on appium@beta from eclipse  and after test failed on real device , screenshot was captured  successfully ^^ no errors.  

I think that for inspecting elements from app  use simulator not real device. so the first problem will be resolved. before new appium desktop version will resleased. 

Main point is that now I am capable  to run tests successfully without any errors. 

Thank you Kazu very much.
